File: src/main/java/de/paulwolf/mandelbrot/core/Mandelbrot.java
package de.paulwolf.mandelbrot.core;

import java.awt.*;

public class Mandelbrot {

    public static int maxIterations = 500;
    public static int contrast = 20;

    public static int mandelbrotIterations(Complex c) {

        Complex v = new Complex(c.real(),c.imaginary());

        for (int i = 0; i < maxIterations; i++) {

            v = v.mandelbrot(c);
            if (v.abs() > 2)
                return i;
        }
        return -1;
    }

    public static int iterationsToColor(int iterations) {

        if (iterations == -1)
            return Color.BLACK.getRGB();
        return new Color(
                Math.max(255 - iterations*contrast, 0),
                Math.max(255 - iterations*contrast, 0),
                Math.max(255 - iterations*contrast, 0),
                255).getRGB();

    }


}

File: src/mandelbrot.cpp
int mandelbrot_iterations(long double cx, long double cy, int max_iterations)
{
    long double x = 0.0L;
    long double y = 0.0L;
    long double xx = 0.0L, yy = 0.0L;

    for (int i = 0; i < max_iterations; i++) {

        // z = z^2 + c
        long double x2 = xx - yy + cx;
        long double y2 = 2.0L * x * y + cy;

        x = x2, y = y2;

        xx = x*x;
        yy = y*y;

        // |z| > 2  <=> |z|^2 > 4
        if (xx+yy > 4.0L)
            return i;
    }
    return max_iterations;
}